vmdb/spec/automation/unit/method_validation/orchestration_check_provisioned_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'spec_helper'
+
+describe "Orchestration check_provisioned Method Validation" do
+  STACK_EMS_REF = "12345"
+  DEPLOY_RESULT = "deploy result"
+  FAILURE_MSG = "failure message"
+
+  let(:ws_url) do
+    "/Cloud/Orchestration/Provisioning/StateMachines/Methods/CheckProvisioned?" \
+    "MiqRequestTask::service_template_provision_task=#{miq_request_task.id}"
+  end
+
+  let(:ws) do
+    MiqAeEngine.instantiate(ws_url)
+  end
+
+  let(:ws_with_refresh_started) do
+    @ae_state   = {'provider_last_refresh' => Time.now, 'deploy_result' => DEPLOY_RESULT}
+    MiqAeEngine.instantiate("#{ws_url}&ae_state_data=#{URI.escape(YAML.dump(@ae_state))}")
+  end
+
+  let(:miq_request_task) do
+    FactoryGirl.create(:miq_request_task,
+                       :destination => service_orchestration,
+                       :miq_request => FactoryGirl.create(:miq_request),
+                       :state       => 'active',
+                       :status      => 'Ok')
+  end
+
+  let(:service_orchestration) do
+    FactoryGirl.create(:service_orchestration,
+                       :orchestration_manager => FactoryGirl.create(:ems_amazon, :last_refresh_date => Time.now - 100))
+  end
+
+  it "waits for the deployment to complete" do
+    ServiceOrchestration.any_instance.stub(:orchestration_stack_status) { ['CREATING', nil] }
```

Comment:
Since we are testing `ServiceOrchestration`, is it possible to not stub this class?
(If it is too much setup, then that is ok. Just something to think about)